packages/mui-base/src/PopperUnstyled/PopperUnstyled.types.ts

```diff
@@ -106,6 +110,8 @@ export interface PopperUnstyledOwnProps {
    * @default false
    */
   transition?: boolean;
+
+  ownerState?: PopperUnstyledOwnerState;
```

Comment:
Added this to pass type check here https://github.com/danhuynhdev/material-ui/blob/2b0f1d9684abd308de1e3c05e8bec4aaa13993f5/packages/mui-base/src/PopperUnstyled/PopperUnstyled.test.tsx#L38
But then I still have to pass `ownerState` as any. Not sure how to properly handle this as I'm not familiar with `ownerState`. From other places, it seems like `ownerState` isn't passed in as a prop but constructed inside the component. But I do not see that in this component.

Reply:
Yeah, something seems to be off here. You should not have `ownerState` as a prop. I'll try to correct the code in the separate PR and you'll be able to rebase on top of my fixes.

Reply:
Have you taken a look a what's wrong here? Happy to fix that myself if I can understand the context here.

Reply:
It turned out to be a bigger problem that affects several components, and we haven't figured out what would be the best solution here.
In short, we generally don't want to expose `ownerState` as a prop, because users of the components should not manually set it. However, in some cases, when an MUI Base component is used in a slot of another component, it will receive the ownerState (as all slot components do). The type of this `ownerState` is **not** `PopperUnstyledOwnerState`, though - it's a state of a component that owns the slots, so it's impossible to type this strongly.

For now, let's define `ownerState?: object` in `PopperUnstyledOwnProps`. When we decide how to solve the problem for good, we'll revisit this part. This should allow you to drop `as any` from tests.

Reply:
Sorry for the late response. I have updated the code but unfortunately, `ownerState?: object` isn't compatible with downstream type, as they expect a concrete interface. So I have to put it as `any` again, i guess it's the same as `object` here as we cannot guarantee typesafe-ness here.